test/instrumentation/modules/@apollo/server.test.js

```diff
@@ -0,0 +1,235 @@
+  resetAgent(done(t, 'HelloQuery hello'))
+  const query = '{"query":"query HelloQuery { hello }"}'
+  initialiseHelloWorldServer(t).then((server) => {
+    const req = http.request(requestOpts('POST'), function (res) {
+      parseResponse(res).then(result => {
+        server.stop()
+        t.strictEqual(result, '{"data":{"hello":"Hello world!"}}\n')
+        agent.flush()
+      })
+    })
+    req.end(query)
+  })
+})
+
+test('POST /graphql - sort multiple queries', function (t) {
+  resetAgent(done(t, 'hello, life'))
+  const typeDefs = `#graphql
+    type Query {
+      hello: String
+      life: Int
+    }
+  `
+
+  const resolvers = {
+    Query: {
+      hello () {
+        t.ok(agent._instrumentation.currTransaction(), 'have active transaction')
+        return 'Hello world!'
+      },
+      life () {
+        t.ok(agent._instrumentation.currTransaction(), 'have active transaction')
+        return 42
+      }
+    }
+  }
+  const query = '{"query":"{ life, hello }"}'
+
+  const server = new ApolloServer({ typeDefs, resolvers })
+  startStandaloneServer(server, { listen: { port: APOLLO_PORT } }).then(() => {
+    const req = http.request(requestOpts('POST'), function (res) {
+      parseResponse(res).then(result => {
+        server.stop()
+        t.strictEqual(result, '{"data":{"life":42,"hello":"Hello world!"}}\n')
+        agent.flush()
+      })
+    })
+    req.end(query)
+  })
+})
+
+test('POST /graphql - sub-query', function (t) {
+  resetAgent(done(t, 'books'))
+  const query = '{"query":"{ books { title author, publisher { name } } }"}'
+  const books = [
+    {
+      title: 'Mikael Hakim',
+      author: 'Mika Waltari',
+      publisher: { name: 'WSOY' }
+    },
+    {
+      title: 'The Life and Times of Scrooge McDuck',
+      author: 'Don Rosa',
+      publisher: { name: 'Egmont' }
+    }
+  ]
+
+  const typeDefs = `#graphql
+    type Publisher {
+      name: String
+    }
+    type Book {
+      title: String
+      author: String
+      publisher: Publisher
+    }
+    type Query {
+      books: [Book]
+    }
+  `
+  const resolvers = {
+    Query: {
+      books () {
+        t.ok(agent._instrumentation.currTransaction(), 'have active transaction')
+        return books
+      }
+    }
+  }
+
+  const server = new ApolloServer({ typeDefs, resolvers })
+  startStandaloneServer(server, { listen: { port: APOLLO_PORT } }).then(() => {
+    const req = http.request(requestOpts('POST'), function (res) {
+      parseResponse(res).then(result => {
+        server.stop()
+        t.strictEqual(result, JSON.stringify({ data: { books } }) + '\n')
+        agent.flush()
+      })
+    })
+    req.end(query)
+  })
+})
+
+function done (t, query) {
+  return function (data, cb) {
+    t.strictEqual(data.transactions.length, 1)
+    t.strictEqual(data.spans.length, 1)
+
+    var trans = data.transactions[0]
+    var span = data.spans[0]
+
+    t.strictEqual(trans.name, query + ' (/graphql)')
```

Comment:
This one fails due to `trans` not appending the request path inside parentheses correctly. I'm somewhat confused by the path appending code in `lib/instrumentation/express-utils.js` since it does not use the `path` property available in an `express` request, yet to my understanding it deals in `express` requests. As seen from line 55, the path truly is `/graphql` in this scenario.

Could this be due to the use of `startStandaloneServer`? As per docs it's using express behind the curtains but maybe it has modified it in such a fashion that the code from `express-utils.js` does not work anymore.

Reply:
I've update the test in commit 81b4120